core/src/main/java/apoc/temporal/TemporalProcedures.java

```diff
@@ -65,15 +67,25 @@ public String formatDuration(
             @Name("input") Object input,
             @Name("format") String format
     ) {
+        
+        DurationValue duration = ((DurationValue) input);
+        
+        final long months = duration.get(MONTHS.name()).value();
+        if (months > 0) {
+            duration = duration
+                    .plus(-1, ChronoUnit.DAYS)
+                    .plus(-1, ChronoUnit.MONTHS);
+        }
+        
         try {
             LocalDateTime midnight = LocalDateTime.of(0, 1, 1, 0, 0, 0, 0);
-            LocalDateTime newDuration = midnight.plus( (DurationValue) input );
+            LocalDateTime newDuration = midnight.plus(duration);
```

Comment:
The initial implementation of this procedure, dating back to 2018, seems to me to be a little obscure and error-prone. Since we are now aware that this piece of code is not working, may I suggest that we take a more holistic look at the problem and see if there isn't a better way to solve this problem which is simpler and more maintainable over the long term?

Cypher's [duration format](https://neo4j.com/docs/cypher-manual/current/functions/temporal/duration/) seems to be structured in a way that would make it easy to parse. 

Since all this APOC procedure does is reformat the Cypher output to a new format, may I suggest that we implement some sort of format-mapper between the two?

It seems like this would be simpler to maintain over the long term than continuing down the path of writing date arithmetic. WDYT?

Reply:
Not sure I really understand what you mean 😅.

You intend, extract the duration components, like `years`, `days`, etc.. , e.g. in this way:
```
 import static org.neo4j.values.storable.DurationFields.*;
 ...
final long years = duration.get(YEARS.propertyKey).value();
final long months = duration.get(MONTHS_OF_YEAR.propertyKey).value();
final long days = duration.get(DAYS.propertyKey).value();
final long hours = duration.get(HOURS.propertyKey).value();
final long minutes = duration.get(MINUTES_OF_HOUR.propertyKey).value();
final long seconds = duration.get(SECONDS_OF_MINUTE.propertyKey).value();
final long nanos = duration.get(NANOSECONDS_OF_SECOND.propertyKey).value();
```

and then apply a sort of `String.format (...)`? Because in this case we cannot use `DateTimeFormatter`s, e.g. `iso_date_time`.

Or maybe place the components in the `LocalDateTime.of(...)`? But in this case we should anyway add `1` at month and days, because it doesn't accept `0`s.

Maybe that's not what you want to implement? Sorry if you mean anything else..

Reply:
@AzuObs I just changed the implementation. The duration is now handled by an ad-hoc implementation, rather than by `DateTimeFormatter`. I made it as consistent as possible with the latter, but obviously some things work differently (e.g. there are no timezone and days of the week)